samples/nrf9160/nrf_cloud_rest_agps/src/main.c

```diff
@@ -0,0 +1,604 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <net/socket.h>
+#include <net/net_ip.h>
+#include <modem/nrf_modem_lib.h>
+#include <modem/at_cmd.h>
+#include <modem/at_notif.h>
+#include <modem/modem_info.h>
+#include <net/tls_credentials.h>
+#include <modem/modem_key_mgmt.h>
+#include <net/http_client.h>
+#include <sys/base64.h>
+#include <net/nrf_cloud_rest.h>
+#include <modem/modem_jwt.h>
+#include <power/reboot.h>
+#include <logging/log.h>
+#include <dk_buttons_and_leds.h>
+#include <net/nrf_cloud_agps.h>
+#include <drivers/gps.h>
+#include <modem/agps.h>
+
+#if defined(CONFIG_REST_ID_SRC_COMPILE_TIME)
+BUILD_ASSERT(sizeof(CONFIG_REST_DEVICE_ID) > 1, "Device ID must be specified");
+#elif defined(CONFIG_REST_ID_SRC_IMEI)
+#define CGSN_RSP_LEN 19
+#define IMEI_LEN 15
+#define IMEI_CLIENT_ID_LEN (sizeof(CONFIG_REST_ID_PREFIX) \
+			    - 1 + IMEI_LEN)
+BUILD_ASSERT(IMEI_CLIENT_ID_LEN <= NRF_CLOUD_CLIENT_ID_MAX_LEN,
+	"REST_ID_PREFIX plus IMEI must not exceed NRF_CLOUD_CLIENT_ID_MAX_LEN");
+#endif
+
+LOG_MODULE_REGISTER(rest_agps_sample, CONFIG_NRF_CLOUD_REST_AGPS_SAMPLE_LOG_LEVEL);
+
+#define REST_RX_BUF_SZ 2100
+#define BUTTON_EVENT_BTN_NUM CONFIG_REST_BUTTON_EVT_NUM
+#define LED_NUM CONFIG_REST_LED_NUM
+#define JITP_REQ_WAIT_SEC 10
+#define NCELL_MEAS_CNT 17
+
+static char device_id[NRF_CLOUD_CLIENT_ID_MAX_LEN + 1];
+
+static K_SEM_DEFINE(button_press_sem, 0, 1);
+static K_SEM_DEFINE(lte_connected, 0, 1);
+static K_SEM_DEFINE(cell_data_ready, 0, 1);
+static K_SEM_DEFINE(do_agps_request_sem, 0, 1);
+
+static struct modem_param_info modem_info;
+static struct lte_lc_cell current_cell;
+
+static const struct device *gps_dev;
+static struct gps_config gps_cfg = {
+	.nav_mode = GPS_NAV_MODE_PERIODIC,
+	.power_mode = GPS_POWER_MODE_DISABLED,
+	.timeout = 120,
+	.interval = 240,
+	.priority = true,
+};
+static struct gps_agps_request agps_request;
+
+#if defined(CONFIG_REST_DO_JITP)
+static bool jitp_requested;
+#endif
+
+void agps_print_enable(bool enable);
+
+static int set_device_id(void)
+{
+	int err = 0;
+
+#if defined(CONFIG_REST_ID_SRC_COMPILE_TIME)
```

Comment:
Consider using the following construct:
```
if (IS_ENABLED(CONFIG_REST_ID_SRC_COMPILE_TIME)) {
	...
} else if (IS_ENABLED(CONFIG_REST_ID_SRC_IMEI)) {
	...
} else if (IS_ENABLED(CONFIG_REST_ID_SRC_INTERNAL_UUID)) {
	...
}
```
This way, the CI will compile each variant regardless of the configuration (unused variants will be optimized out from the final binary). Unless there are other configuration constraints this is preferred way to handle such cases.

Reply:
This section will be replaced with a simple function call once https://github.com/nrfconnect/sdk-nrf/pull/5730 is merged.